[CI] AnalyzerTests testMvAppendValidation failing

**Build Scans:**
- [elasticsearch-intake #11942 / part3](https://gradle-enterprise.elastic.co/s/grqenhvcs5b6k)
- [elasticsearch-intake #11939 / part3](https://gradle-enterprise.elastic.co/s/yrqaqogrbzbfo)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.analysis.AnalyzerTests.testMvAppendValidation" -Dtests.seed=3FBA8F4231345F26 -Dtests.locale=en-SE -Dtests.timezone=America/Denver -Druntime.java=23
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.analysis.AnalyzerTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testMvAppendValidation),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a string containing "second argument of [mv_append(text, unsigned_long)] must be [text], found value [unsigned_long] type [unsigned_long]"
     but: was "Found 1 problem
line 1:19: second argument of [mv_append(text, unsigned_long)] must be [keyword], found value [unsigned_long] type [unsigned_long]"
```

**Issue Reasons:**
- [8.x] 2 consecutive failures in test testMvAppendValidation
- [8.x] 2 consecutive failures in step part3
- [8.x] 2 consecutive failures in pipeline elasticsearch-intake
- [8.x] 2 failures in test testMvAppendValidation (1.2% fail rate in 164 executions)
- [8.x] 2 failures in step part3 (5.7% fail rate in 35 executions)
- [8.x] 2 failures in pipeline elasticsearch-intake (5.7% fail rate in 35 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -282,9 +282,6 @@ tests:
 - class: org.elasticsearch.oldrepos.OldRepositoryAccessIT
   method: testOldRepoAccess
   issue: https://github.com/elastic/elasticsearch/issues/115631
-- class: org.elasticsearch.xpack.esql.analysis.AnalyzerTests
-  method: testMvAppendValidation
-  issue: https://github.com/elastic/elasticsearch/issues/115636
 
 # Examples:
 #

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/AnalyzerTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/AnalyzerTests.java
@@ -56,6 +56,7 @@ import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.function.Function;
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
@@ -1879,6 +1880,7 @@ public class AnalyzerTests extends ESTestCase {
         Supplier<Integer> supplier = () -> randomInt(fields.length - 1);
         int first = supplier.get();
         int second = randomValueOtherThan(first, supplier);
+        Function<String, String> noText = (type) -> type.equals("text") ? "keyword" : type;
 
         String signature = "mv_append(" + fields[first][0] + ", " + fields[second][0] + ")";
         verifyUnsupported(
@@ -1886,7 +1888,7 @@ public class AnalyzerTests extends ESTestCase {
             "second argument of ["
                 + signature
                 + "] must be ["
-                + fields[first][1]
+                + noText.apply(fields[first][1])
                 + "], found value ["
                 + fields[second][0]
                 + "] type ["

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/AnalyzerTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/AnalyzerTests.java
@@ -1881,6 +1881,10 @@ public class AnalyzerTests extends ESTestCase {
         int first = supplier.get();
         int second = randomValueOtherThan(first, supplier);
         Function<String, String> noText = (type) -> type.equals("text") ? "keyword" : type;
+        assumeTrue(
+            "Ignore tests with TEXT and KEYWORD combinations because they are now valid",
+            noText.apply(fields[first][0]).equals(noText.apply(fields[second][0])) == false
+        );
 
         String signature = "mv_append(" + fields[first][0] + ", " + fields[second][0] + ")";
         verifyUnsupported(

